include/deal.II/lac/diagonal_matrix.h

```diff
@@ -179,6 +179,12 @@ class DiagonalMatrix : public Subscriptor
                    const VectorType &src) const;
 
   /**
+   * Initialize vector @p dst. This is a part of the interface required
+   * by linear operator.
```

Comment:
probably @tamiko knows better what this should do, but I say it's only initialisation to be of correct size.